src/ML-SNAP/compute_sna_grid.cpp

```diff
@@ -0,0 +1,328 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/ Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "compute_sna_grid.h"
+
+#include "atom.h"
+#include "comm.h"
+#include "domain.h"
+#include "error.h"
+#include "force.h"
+#include "memory.h"
+#include "modify.h"
+#include "neigh_list.h"
+#include "neigh_request.h"
+#include "neighbor.h"
+#include "pair.h"
+#include "sna.h"
+#include "tokenizer.h"
+#include "update.h"
+
+#include <cmath>
+#include <cstring>
+
+using namespace LAMMPS_NS;
+
+ComputeSNAGrid::ComputeSNAGrid(LAMMPS *lmp, int narg, char **arg) :
+    ComputeGrid(lmp, narg, arg), cutsq(nullptr), radelem(nullptr), wjelem(nullptr)
+{
+  double rfac0, rmin0;
+  int twojmax, switchflag, bzeroflag, bnormflag, wselfallflag;
+
+  // skip over arguments used by base class
+  // so that argument positions are identical to
+  // regular per-atom compute
+
+  arg += nargbase;
+  narg -= nargbase;
+
+  int ntypes = atom->ntypes;
+  int nargmin = 6 + 2 * ntypes;
+
+  if (narg < nargmin) error->all(FLERR, "Illegal compute sna/grid command");
+
+  // default values
+
+  rmin0 = 0.0;
+  switchflag = 1;
+  bzeroflag = 1;
+  quadraticflag = 0;
+  chemflag = 0;
+  bnormflag = 0;
+  wselfallflag = 0;
+  switchinnerflag = 0;
+  nelements = 1;
+
+  // process required arguments
+
+  memory->create(radelem, ntypes + 1, "sna/grid:radelem");    // offset by 1 to match up with types
+  memory->create(wjelem, ntypes + 1, "sna/grid:wjelem");
+
+  rcutfac = atof(arg[3]);
```

Comment:
calls to `atof()` should be replaced with calls to utils::numeric(), and `atoi()` with utils::inumeric(), respectively.

Reply:
> calls to `atof()` should be replaced with calls to utils::numeric(), and `atoi()` with utils::inumeric(), respectively.

Done. Also done for the 4 other SNAP compute, now they are all the same. I will submit those changes in a separate PR, I think.

Reply:
They would fit in here just fine.